lib/server.js

```diff
@@ -264,7 +264,34 @@ function Client(stream, socket) {
   }
   // end service/auth-related ==================================================
 
+  var unrepliedGlobalRequests = [];
+
+  function sendReplies() {
+    var reply;
+    while (unrepliedGlobalRequests[0] && unrepliedGlobalRequests[0].type) {
+      reply = unrepliedGlobalRequests.shift();
+      if (reply.type === 'SUCCESS') {
+        stream.requestSuccess(reply.buf);
+      }
+      if (reply.type === 'FAILURE') {
+        stream.requestFailure();
+      }
+    }
+  }
+
   stream.on('GLOBAL_REQUEST', function(name, wantReply, data) {
+    var reply = {};
```

Comment:
Minor performance nit: can you predefine the properties in the object literal here? Just set them initially to `null`.